keras_nlp/layers/transformer_decoder.py

```diff
@@ -36,6 +36,11 @@ class TransformerDecoder(keras.layers.Layer):
     [guide](https://keras.io/guides/understanding_masking_and_padding/)
     for more details.
 
+    If decoder_only is set to True, the encoder layer would not be built, 
```

Comment:
We have removed this argument. We should remove docs too.

We should update the class level docs with a few things

 - In the second paragraph about masking. Add as a first sentence, "This layer will always apply a causal mask to the decoder attention layer."
 - Add a new paragraph. Some suggested text below:

```
This layer can be called with with either one or two inputs as follows:

 - `layer(decoder_sequence)`: no cross-attention will be built into the decoder
   block. This is useful when building a "decoder-only" transformer such as GPT-2.
 - `layer(decoder_sequence, encoder_sequence)`: cross-attention will be built into
   the encoder block. This is useful when building an "encoder-decoder" transformer,
   such as the original transformer model described in Attention is All You Need.